packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -350,6 +371,39 @@ module.exports = {
     // https://github.com/jmblog/how-to-optimize-momentjs-with-webpack
     // You can remove this if you don't use Moment.js:
     new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
+
+    // For some reason, Webpack adds some ids of all the modules that exist to our vendor chunk
+    // Instead of using numerical ids it uses a unique path to map our request to a module.
+    // Thanks to this change the vendor hash will now always stay the same
+    new webpack.NamedModulesPlugin(),
+
+    // Ensure that every chunks have an actual name and not an id
+    // If the chunk has a name, this name is used
+    // otherwise the name of the file is used
+    new webpack.NamedChunksPlugin(chunk => {
+      if (chunk.name) {
+        return chunk.name;
+      }
+      return chunk
+        .mapModules(m => m.request)
+        .map(filename => path.basename(filename, path.extname(filename)))
+        .join('_');
+    }),
+
+    // Avoid having the vendors in the rest of the app
+    new webpack.optimize.CommonsChunkPlugin({
+      name: 'vendors',
```

Comment:
Do we need to exclude this commons chunk if we don't specify `vendors` in `entry`? Or is it simply ignored?

Reply:
I've rechecked, and if the file exists, the vendor entry weights 40KB, and the main 10KB (can't remember the exact values), and without, webpack compiles and there is no vendors entry and main weighs 50KB

Reply:
I'd like to exclude this for safety reasons; we're not sure how webpack might regress in the future.

Reply:
```javascript
fs.existsSync(paths.appVendors) ? new webpack.optimize.CommonsChunkPlugin({
      name: 'vendors',
      minChunks: Infinity,
    }) : null,
```
is enough ? (i've tested and it's working)

Reply:
```javascript
...(() => fs.existsSync(paths.appVendors) ?
  [new webpack.optimize.CommonsChunkPlugin({
      name: 'vendors',
      minChunks: Infinity,
    })] : [])(),
```

or this ?